pyccel/ast/datatypes.py

```diff
@@ -81,7 +81,37 @@
     "logical" : {
         4  : "C_BOOL"}
 }
-
+iso_c_binding_shortcut = {
+    "integer" : {
+        1  : 'i8',
+        2  : 'i16',
+        4  : 'i32',
+        8  : 'i64',
+        16 : 'i128'}, #no supported yet
+    "real"    : {
+        4  : 'f32',
+        8  : 'f64',
+        16 : 'f128'},
+    "logical" : {
+        4  : "b4"}
+}
+iso_c_bindings = ['C_INT8_T', 'C_INT16_T', 'C_INT32_T', 'C_INT64_T', 'C_INT128',
+                  'C_FLOAT', 'C_DOUBLE', 'C_LONG_DOUBLE', "C_BOOL"]
+iso_c_binding_shortcuts = ['i8', 'i16', 'i32', 'i64', 'i128', 'f32', 'f64', 'f128', "b4"]
```

Comment:
iso_c_binding_shortcut is used by print_kind to find the shortcut to use based on expr.dtype and expr.precision, whereas  iso_c_binding_shortcuts is used to print the use statement and it does not have access to the expr variable. The class keeps a boolean array of the shortcuts used and i thought it would be good to retrieve the name and shortcut from an array of strings instead of the dictionary that requires expr for look up.

Reply:
I don't think the dictionaries are costly enough to be a problem. Having 2 objects with the same contents is usually a bad idea though (see [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle).

>  shall i rely solely on the dictionaries?

Personally I think I would, it's clear and requires fewer modifications if you add a new datatype

> whereas iso_c_binding_shortcuts is used to print the use statement and it does not have access to the expr variable

You could also save a set of all the objects you used from `iso_c_binding_shortcut`